src/borda_clj/client.clj

```diff
@@ -0,0 +1,61 @@
+(ns borda-clj.client
+  (:use [slingshot.slingshot :only [throw+ try+]])
+  (:require [aleph.http :as http]
+            [byte-streams :as bs]
+            [cheshire.core :as json]))
+
+(defn collect
+  "Adds the given measurement (values and dimensions) to the given hashmap and caps the size of the hashmap at the given max-buffer-size"
+  [measurements max-buffer-size dimensions values]
+  (let [key (into (sorted-map) dimensions)]
+    (if (contains? measurements key)
+      ; TODO: support stuff other than SUM here (e.g. AVG, MIN, MAX, etc.)
+      (update measurements key (partial merge-with +) values) ; merge with existing in buffer
+      (if (< (count measurements) max-buffer-size)
+        (assoc measurements key values) ; space available, add to buffer
+        (do (print "borda buffer full, discarding measurement" dimensions values)
+            measurements))))) ; buffer full
+
+(defn reducing-submitter
+  "Returns two functions. The first is a reducing submitter that collects and
+   aggregates measurements and sends them using the given send function at the
+   specified interval (in milliseconds) and limits the size of pending
```

Comment:
Actual interval is the specified interval plus whatever `send` takes.  I don't think this is bad, but it's good to keep in mind this detail.